lib/pdf_fill/forms/form_helper.rb

```diff
@@ -76,6 +76,16 @@ def split_date(date)
         }
         split_date
       end
+
+      def combine_extra_date_ranges(date_range_array)
+        return if date_range_array.nil?
+        extras_ranges = []
+        date_range_array.each do |range|
```

Comment:
I think you are suggesting that we do something like: 

"select everything from this array that isn't null"
date_range_array.select{|date_range| date_range.nil?}.each do |range|
          extras_ranges.push('from: ' + range['from'] + ' to: ' + range['to'])
end

instead of

date_range_array.each do |range|
          next if range.nil?
          extras_ranges.push('from: ' + range['from'] + ' to: ' + range['to']) 
end

Is that what you mean?  In this case, is the select really more readable?  Can you elaborate?  Thank you.

Reply:
`select` was probably a poor suggestion, but I think with `map` you could do something like
```
date_range_array.map! do |range|
  unless range.nil?
    'from: ' + range['from'] + ' to: ' + range['to']
  end
end.compact
```
In the end I think this is more of a stylistic choice so feel free to ignore me :)

Reply:
That's great, thank you!  I'll make that change.

Reply:
@egarreau @hopelam : Let's keep as it is right now if that's acceptable. Else we will have to split the places where this function is called to split / convert string to array based on a delimiter.

Reply:
This would work without any other code changes, but not sure what value it would add. Spec to test --> spec/lib/pdf_fill/filler_spec.rb

    def combine_extra_date_ranges(date_range_array)
        return if date_range_array.nil?
        extras_ranges = []
        date_range_array.map do |range|
          extras_ranges.push('from: ' + range['from'] + ' to: ' + range['to']) unless range.nil?
        end
        extras_ranges.join("\n")
      end

Reply:
getting into code golf territory 😄 but does this work?
```ruby
def combine_extra_date_ranges2(date_range_array)
  return if date_range_array.nil?
  date_range_array.map { |r| r = 'from: ' + r['from'] + ' to: ' + r['to'] if r }.join("\n")
end
```

Reply:
it does work, makes it stylistic but may be not simple to read :-) I will push the changes with some minor fixes

    def combine_date_ranges(date_range_array)
        return if date_range_array.nil?
        date_range_array.map { |r| 'from: ' + r['from'] + ' to: ' + r['to'] if r }.join("\n")
      end

Reply:
Completed the changes and push.  Thanks